File: BaekJoon/DP/1932.py
import sys


N = int(sys.stdin.readline())

tri = []

for _ in range(N):
    tri.append(list(map(int, sys.stdin.readline().split())))

result = [tri[-1]]

tri = tri[::-1]

for floor in range(N-1):
    temp = []
    for idx in range(N-1-floor):
        value = tri[floor+1][idx] + max(result[floor][idx], result[floor][idx+1])

        temp.append(value)
    result.append(temp)

print(result[-1][0])

File: BaekJoonSwift/BaekJoonSwift/DP/1932.swift
import Foundation

var cost = [[Int]]()
let n = Int(readLine()!)!

for _ in 0..<n {
    cost.append(readLine()!.split(separator: " ").map { Int($0)! })
}

if n == 1 {
    print(cost[0][0])
} else {
    for i in 2...n {
        for j in 0..<cost[n-i].count {
            cost[n-i][j] += max(cost[n-i+1][j], cost[n-i+1][j+1])
        }
    }
    print(cost[0][0])
}
